src/full/Agda/Syntax/Abstract.hs

```diff
@@ -1104,3 +1104,99 @@ insertImplicitPatSynArgs wild r ns as = matchArgs r ns as
     matchArgs r (n:ns) as = do
       (p, as) <- matchNextArg r n as
       first ((unArg n, p) :) <$> matchArgs (getRange p) ns as
+
+------------------------------------------------------------------------
+-- Declaration spines
+------------------------------------------------------------------------
+
+-- | Declaration spines. Used in debugging to make it easy to see
```

Comment:
If declaration spines are only used for debugging, I like to see them placed in a separate module like `Syntax.Abstract.Debug` so that a newcomer / other developer knows they can ignore them when trying to understand the codebase.

Reply:
My guess is that it takes about as much time to skip that file as it takes to skip this section of this file, but I don't mind if you refactor the code. The following code, which seems to have been initially written by you, could perhaps also be moved to the new file:

https://github.com/agda/agda/blob/d50b1e967818974b2af797c2641d2c8f9db1ede9/src/full/Agda/TypeChecking/Rules/Decl.hs#L1018-L1062